fork-common/src/main/java/com/shazam/fork/suite/TestSuiteLoader.java

```diff
@@ -107,10 +104,42 @@ private boolean isMethodIgnored(AnnotationItem... annotationItems) {
                 .map(annotationItem -> annotationItem.getEncodedAnnotation().values)
                 .flatMap(encodedValues -> stream(encodedValues)
                         .flatMap(encodedValue -> stream(((ArrayEncodedValue) encodedValue).values)
-                                .map(stringEncoded -> ((StringEncodedValue)stringEncoded).value.getStringValue())))
+                                .map(stringEncoded -> ((StringEncodedValue) stringEncoded).value.getStringValue())))
+                .collect(toList());
+    }
+
+    private Map<String, String> getTestProperties(AnnotationItem[] annotations) {
+        Map<String, String> properties = new HashMap<>();
+        stream(annotations)
+                .filter(annotationItem -> TEST_PROPERTIES_ANNOTATION.equals(stringType(annotationItem)))
+                .map(AnnotationItem::getEncodedAnnotation)
+                .forEach(an -> {
+                    List<String> keys = getAnnotationProperty(an, "keys");
+                    List<String> values = getAnnotationProperty(an, "values");
+
+                    for (int i = 0; i < min(values.size(), keys.size()); i++) {
+                        properties.put(keys.get(i), values.get(i));
+                    }
+                });
+        return properties;
+    }
+
+    private List<String> getAnnotationProperty(AnnotationEncodedSubValue an, String propertyName) {
+        EncodedValue[] values = ((ArrayEncodedValue) an.values[indexOfName(an, propertyName)]).values;
+        return stream(values)
+                .map(stringEncodedValue -> ((StringEncodedValue) stringEncodedValue).value.getStringValue())
                 .collect(toList());
     }
 
+    private int indexOfName(AnnotationEncodedSubValue p, String key) {
```

Comment:
What about adding some form of protection or clearer output message in case of empty annotation?
If for example someone annotates a test with:
`    @TestProperties( )`  
the output only says
`Error while Fork was executing
java.lang.ArrayIndexOutOfBoundsException: -1
        at com.shazam.fork.suite.TestSuiteLoader.getAnnotationProperty(TestSuiteLoader.java:128)`

Maybe a more explicit error message would make sense.

Reply:
Fair point about the -1, I'll just make it ignore such annotations but not log anything as it's valid input from a "this is a properties map" perspective. 

About the second point, I wouldn't think it's big enough to split but I can fix if we have reviewer consensus.